File: lab9/shoes.py
class Shoes:
    """
    Абстрактний клас Shoes, що описує основні характеристики та функціональність взуття.
    """

    def __init__(self, brand, size, color):
        """
        Конструктор для створення взуття.

        :param brand: Бренд взуття.
        :param size: Розмір взуття.
        :param color: Колір взуття.
        """
        self.brand = brand
        self.size = size
        self.color = color
        self.laced = False
        self.worn = False
        self.distance_run = 0.0

    def needs_repair(self):
        """
        Перевіряє, чи потрібно ремонтувати взуття.

        :return: True, якщо пробіг >= 100 км.
        """
        return self.distance_run >= 100

    def add_distance(self, distance):
        """
        Додає дистанцію до загального пробігу.

        :param distance: Пробіг (км).
        """
        self.distance_run += distance

    # Геттери та сеттери для атрибутів класу

    def get_brand(self):
        return self.brand

    def get_size(self):
        return self.size

    def get_color(self):
        return self.color

    def is_laced(self):
        return self.laced

    def set_laced(self, laced):
        self.laced = laced

    def is_worn(self):
        return self.worn

    def set_worn(self, worn):
        self.worn = worn

    def get_distance_run(self):
        return self.distance_run

File: Lab3/src/Ki_306/Petruk/Lab3/Shoes.java
// File: Ki_306/Petruk/Lab3/Shoes.java
package Ki_306.Petruk.Lab3;

/**
 * Абстрактний клас Shoes, що описує основні характеристики та функціональність взуття.
 */
public abstract class Shoes {
    private String brand;
    private double size;
    private String color;
    private boolean laced;
    private boolean worn;
    private double distanceRun;

    /**
     * Конструктор для створення взуття.
     *
     * @param brand Бренд взуття.
     * @param size Розмір взуття.
     * @param color Колір взуття.
     */
    public Shoes(String brand, double size, String color) {
        this.brand = brand;
        this.size = size;
        this.color = color;
        this.laced = false;
        this.worn = false;
        this.distanceRun = 0.0;
    }

    /**
     * Перевіряє, чи потрібно ремонтувати взуття.
     *
     * @return true, якщо взуття пробігло 100 км або більше, і потребує ремонту.
     */
    public boolean needsRepair() {
        return distanceRun >= 100;
    }

    /**
     * Додає дистанцію пробігу до загального пробігу взуття.
     *
     * @param distance Дистанція, що була пробігла.
     */
    protected void addDistance(double distance) {
        this.distanceRun += distance;
    }

    // Геттери та сеттери для атрибутів класу

    public String getBrand() {
        return brand;
    }

    public double getSize() {
        return size;
    }

    public String getColor() {
        return color;
    }

    public boolean isLaced() {
        return laced;
    }

    protected void setLaced(boolean laced) {
        this.laced = laced;
    }

    public boolean isWorn() {
        return worn;
    }

    protected void setWorn(boolean worn) {
        this.worn = worn;
    }

    public double getDistanceRun() {
        return distanceRun;
    }
}
